core/lib/spree/core/engine.rb

```diff
@@ -105,6 +105,12 @@ class Engine < ::Rails::Engine
         ]
       end
 
+      initializer 'spree.promo.register.promotions.shipping_actions', before: :load_config_initializers do |app|
```

Comment:
A couple other possibilities:

A) Just decide we don't care about the scenario above

B) Figure out how to make shipping promotions work correctly without the `before_transition from: :delivery, do: :apply_shipping_promotions` transition (thus eliminating the need to identify them in `Spree::PromotionHandler::Shipping`).

C) Think of another way to identify Shipping promo actions in `Spree::PromotionHandler::Shipping`.  Perhaps all Shipping promo actions must inherit from a base class?  Or have a `is_shipping` field on the action?

It might be worth just going w/ what's in this PR and iterating on it?